feat: add Module Site Deployment workflow (#59)
MovingBlocks/ModuleSite: docs/project-deployment.md

# ModuleSite Deployment

> [ModuleSite](https://github.com/MovingBlocks/ModuleSite) uses a [Github Actions](https://github.com/features/actions) to automate the deployment process.On every push Github actions are triggered and ModuleSite is build and it is deployed on

Reviewer: ```suggestion
[ModuleSite](https://github.com/MovingBlocks/ModuleSite) uses a [Github Action](https://github.com/features/actions) to automate the deployment process. On every push to `master`, the GitHub action is triggered, builds the ModuleSite and deploys it on
```

> [Github pages](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages)

Reviewer: ```suggestion
[GitHub pages](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages)
```

## Requirements and Working

> <b>Gatsby Publish:</b> Github Action to build and deploy Gatsby site on Github pages.More about [Gatsby Publish](https://github.com/marketplace/actions/gatsby-publish)

Reviewer: In markdown, you can use `**some bold text**` to get "**some bold text**"

Reviewer: ```suggestion
<b>[Gatsby Publish](https://github.com/marketplace/actions/gatsby-publish):</b> Github Action to build and deploy Gatsby site on Github pages.
```

> Everytime the Github actions are triggered it executes `gatsby build` at the root of repository and deploy it to github pages.

Reviewer: ```suggestion
Everytime the GitHub action is triggered, it executes `gatsby build` at the root of the repository and deploys it to GitHub pages.
```

You can find our Github action workflow in `.github\workflow`

Before using Action their are some configuration options that are provide by Gatsby Publish

- <b>access-token: </b>
  A GitHub Personal Access Token with access of pushing and creating pull request.This is require to push builds after building the Site.The access token should be stored as secret text in the repository settings

- <b>deploy-branch: </b>
  This will be branch where all the outputs after executing `gatsby build` will be push. This includes static files that were generated duing build process

- <b>gatsby-args: </b>

>   Additional arguments that get passed to gatsby build. See the [Gatsby documentation](https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/path-prefix/) for a list of allowed options. Provided as an input. Defaults to nothing.

Reviewer: This is something that should be rather done as comments in the workflow yaml.

# ModuleSite Deployment

[ModuleSite](https://github.com/MovingBlocks/ModuleSite) uses a [Github Action](https://github.com/features/actions) to automate the deployment process. On every push to `master`, the GitHub action is triggered, builds the ModuleSite and deploys it on
[GitHub pages](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages)

## Requirements

**[Gatsby Publish](https://github.com/marketplace/actions/gatsby-publish):** Github Action to build and deploy Gatsby site on Github pages.

## Working

Everytime the GitHub action is triggered, it executes `gatsby build` at the root of the repository and deploys it to GitHub pages.
You can find our Github action workflow in `.github\workflow`

> Before using Action their are some configuration options that are provide by Gatsby Publish

Reviewer: ```suggestion
For using the GitHub action, Gatsby Publish provides some configuration options:
```

- **access-token:**

>   A GitHub Personal Access Token with access of pushing and creating pull request.This is require to push builds after building the Site.The access token should be stored as secret text in the repository settings

Reviewer: ```suggestion
  A GitHub Personal Access Token with access for pushing and creating pull requests. This is required to push builds after building the Site. The access token should be stored as a secret in the repository settings.
```

- **deploy-branch:**

>   This will be branch where all the outputs after executing `gatsby build` will be push. This includes static files that were generated duing build process

Reviewer: ```suggestion
  This is the branch to which all the outputs after executing `gatsby build` will be pushed. This includes static files that were generated during build process.
```

- **gatsby-args:**

>   Additional arguments that get passed to gatsby build. See the [Gatsby documentation](https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/path-prefix/) for a list of allowed options. Provided as an input. Defaults to nothing.

Reviewer: ```suggestion
  Additional arguments that get passed to `gatsby build`. See the [Gatsby documentation](https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/path-prefix/) for a list of allowed options. Provided as an input. Defaults to nothing.
```

> The above configuration options were used by ModuleSite.For more information about configuration visit Gatsby Publish

Reviewer: ```suggestion
The above configuration options are used by the ModuleSite. For more information about the configuration visit Gatsby Publish.
```

Reviewer: > visit Gatsby Publish

I guess you mean the documentation of "Gatsby Publish"? Please add a link here.

Author: done

> > **⚠️ NOTE:** The Gatsby code is on root directory so after build process `./public` directory is generated.Gatsby do not allow to edit or customize the public directory, so we have to build code everytime when their is any changes to Gatsby code.Gatsby automatically provides a build script on `Package.json` which is required by the Action to execute the build process

Reviewer: ```suggestion
> **⚠️ NOTE:** The Gatsby code is located in the root directory. After the build process the `./public` directory is generated. Gatsby does not allow to edit or customize the public directory, so we have to build code every time the Gatsby code changes. Gatsby automatically provides a build script on `Package.json` which is required by the GitHub action to execute the build process.
```

Author: done all the changes
